core/drand_beacon_control.go

```diff
@@ -1138,22 +1137,120 @@ func (bp *BeaconProcess) StartFollowChain(req *drand.StartFollowRequest, stream
 	defer cbStore.RemoveCallback(addr)
 
 	syncer := beacon.NewSyncManager(&beacon.SyncConfig{
-		Log:      bp.log,
-		Store:    cbStore,
-		Info:     info,
-		Client:   bp.privGateway,
-		Clock:    bp.opts.clock,
-		NodeAddr: bp.priv.Public.Address(),
+		Log:         bp.log,
+		Store:       cbStore,
+		BoltdbStore: store,
+		Info:        info,
+		Client:      bp.privGateway,
+		Clock:       bp.opts.clock,
+		NodeAddr:    bp.priv.Public.Address(),
 	})
-	go syncer.Run()
+	go syncer.Run(ctx)
 	defer syncer.Stop()
-	syncer.RequestSync(peers, req.GetUpTo())
+
+	for {
+		syncer.RequestSync(req.GetUpTo(), peers)
+		// wait for all the callbacks to be called and progress sent before returning
+		select {
+		case <-done:
+			return nil
+		case <-ctx.Done():
+			return ctx.Err()
+		// we re-send a sync request every 10 periods in case the process got staled and let the sync manager handle
```

Comment:
Why would this happen ? It seems to me like a workaround around something that should work. did you see it happen that the sync manager gets stalled ? it should not...

Reply:
Ok got it -thanks that makes sense.